addons/koji/jaxrs/src/main/java/org/commonjava/indy/koji/bind/jaxrs/KojiRepairResource.java

```diff
@@ -0,0 +1,102 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.koji.bind.jaxrs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import org.commonjava.indy.bind.jaxrs.IndyResources;
+import org.commonjava.indy.bind.jaxrs.SecurityManager;
+import org.commonjava.indy.koji.data.KojiRepairException;
+import org.commonjava.indy.koji.data.KojiRepairManager;
+import org.commonjava.indy.koji.model.KojiRepairRequest;
+import org.commonjava.indy.koji.model.KojiRepairResult;
+import org.commonjava.indy.model.core.PackageTypeDescriptor;
+import org.commonjava.indy.model.core.PackageTypes;
+import org.commonjava.indy.util.ApplicationContent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriInfo;
+
+import static org.commonjava.indy.bind.jaxrs.util.ResponseUtils.throwError;
+import static org.commonjava.indy.koji.IndyKojiConstants.REPAIR_KOJI;
+import static org.commonjava.indy.koji.IndyKojiConstants.VOL;
+import static org.commonjava.indy.util.ApplicationContent.application_indy_star_json;
+import static org.commonjava.indy.util.ApplicationContent.application_json;
+
+@Api( value = "Koji repairVolume operation", description = "Repair Koji remote repositories." )
+@Path( "/api/" + REPAIR_KOJI )
+@Produces( { application_json, application_indy_star_json } )
+public class KojiRepairResource
+                implements IndyResources
+{
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    private ObjectMapper mapper;
+
+    @Inject
+    private SecurityManager securityManager;
+
+    @Inject
+    private KojiRepairManager repairManager;
+
+    @ApiOperation( "Repair koji repository remote url /vol." )
+    @ApiResponse( code = 200, message = "Operation finished (consult response content for success/failure).",
+                    response = KojiRepairResult.class )
+    @ApiImplicitParam( name = "body", paramType = "body",
+                    value = "JSON request specifying source and other configuration options",
+                    required = true, dataType = "org.commonjava.indy.koji.model.KojiRepairRequest" )
+    @POST
+    @Path( "/" + VOL )
```

Comment:
I thought to separate via REST endpoint to give the user more control over which to repair. If we do other repairs in future, we may not want to go through /vol part that has been done. so if we make it general, we have to stack out /vol logic and stack in the other stuff. This will involve some code change in the "stack" - you have to decide which are in the stack which not, at a certain time. OTOH, if using different endpoint, we just add new endpoint/code. This is namely "addition vs modification" of code. I'd prefer the former. 
Using different endpoint also makes it clear to users so they know what they are doing. 
But this design is arguable. I am open to advise.